File: numbers/prob1342.py
class Solution(object):
    def numberOfSteps (self, num):
        """
        :type num: int
        :rtype: int
        """
        curr = num
        i    = 0
        while curr > 0:
            if curr & 1:
                curr -= 1
            else:
                curr //= 2
  				# curr = curr >> 1
            i += 1
        return i

File: math/prob1342.kt
// 1342 Number of Steps to Reduce a Number to Zero 
//
// class Solution {
    fun numberOfSteps(num: Int): Int {

        var steps = 0
        var tmp = num
        while (tmp > 0) {
            steps++
            tmp = if (tmp % 2 == 0) tmp/2 else tmp-1
        }

        return steps
    }
}
